app/com/microsoft/azure/iotsolutions/devicetelemetry/services/StorageClient.java

```
@@ -3,53 +3,136 @@
                 ConnectionPolicy.GetDefault(),
                 ConsistencyLevel.Session);
 
-            if (documentClient == null) {
+            if (this.client == null) {
                 // TODO add logging if connection fails (don't log connection string)
+                log.error("Could not connect to DocumentClient");
                 throw new InvalidConfigurationException("Could not connect to DocumentClient");
             }
         }
 
-        return documentClient;
+        return this.client;
     }
 
+    @Override
+    public ResourceResponse<DocumentCollection> createCollectionIfNotExists(String id) {
+        DocumentCollection collectionInfo = new DocumentCollection();
+        RangeIndex index = Index.Range(DataType.String, -1);
+        collectionInfo.setIndexingPolicy(new IndexingPolicy(new Index[]{ index }));
+        collectionInfo.setId(id);
+
+        // Azure Cosmos DB collections can be reserved with throughput specified in request units/second.
+        // Here we create a collection with 400 RU/s.
+        RequestOptions requestOptions = new RequestOptions();
+        requestOptions.setOfferThroughput(400);
+        String dbUrl = "/dbs/" + this.telemetryDb.getId();
+        String colUrl = dbUrl + "/colls/" + id;
+        boolean create = false;
+        ResourceResponse<DocumentCollection> response = null;
+
+        try {
+            response = this.client.readCollection(colUrl, requestOptions);
+        } catch (DocumentClientException dcx) {
+            if (dcx.getStatusCode() == Http.Status.NOT_FOUND) {
+                create = true;
+            } else {
+                log.error("Error reading collection: " + id, dcx);
+            }
+        }
+
+        if (create) {
+            try {
+                response = this.client.createCollection(dbUrl, collectionInfo, requestOptions);
+            } catch (DocumentClientException dcx) {
+                log.error("Error creating collection: " + id, dcx);
```

I'd catch "Exception", i.e. network issues too.
Should we rethrow rather than swallowing the exceptions?

(same comment in other methods)